calico_containers/tests/st/test_base.py

```diff
@@ -20,3 +24,26 @@ def tearDown(self):
         containers = docker.ps("-qa").split()
         for container in containers:
             DockerHost.delete_container(container)
+
+    def start_etcd(self):
+        """
+        Starts a separate etcd container.
+        """
+
+        docker.run("-d",
+                   "-p", "2379:2379",
+                   "-p", "2380:2380",
+                   "--name", "etcd", "quay.io/coreos/etcd:v2.0.10",
+                   name="calico",
+                   advertise_client_urls="http://%s:2379" % self.ip,
+                   listen_client_urls="http://0.0.0.0:2379",
+                   initial_advertise_peer_urls="http://%s:2380" % self.ip,
+                   listen_peer_urls="http://0.0.0.0:2380",
+                   initial_cluster_token="etcd-cluster-2",
+                   initial_cluster="calico=http://%s:2380" % self.ip,
+                   initial_cluster_state="new",
+                  )
+
+    def get_ip(self):
```

Comment:
just butting in here too.  not convinced about assuming anything about the host interfaces here.  An oft touted way to do this is along the lines of:

import socket
s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
s.connect(('google.com', 0))
s.getsockname()[0]

not sure that's much better, but it doesn't assume eth0.